staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -293,13 +296,22 @@ func (c *dynamicResourceClient) Get(ctx context.Context, name string, opts metav
 	return uncastObj.(*unstructured.Unstructured), nil
 }
 
-func (c *dynamicResourceClient) List(ctx context.Context, opts metav1.ListOptions) (result *unstructured.UnstructuredList, err error) {
-	defer func() {
+func (c *dynamicResourceClient) List(ctx context.Context, opts metav1.ListOptions) (*unstructured.UnstructuredList, error) {
```

Comment:
for easier review, this method should match the one we have in client-go, the PR that added this functionality can be found at https://github.com/kubernetes/kubernetes/pull/124509

Reply:
For easier review, please see this 5000 line PR. ;)